src/java.base/share/classes/java/lang/invoke/ConstantBootstraps.java

```diff
@@ -373,13 +373,14 @@ public static VarHandle arrayVarHandle(MethodHandles.Lookup lookup, String name,
      *     is applied to {@code value} as if by calling {@code dstType.cast(value)}.
      * <li>If {@code dstType} is a primitive type, then, if the runtime type
      *     of {@code value} is a primitive wrapper type (such as {@link Integer}),
-     *     a Java unboxing conversion is applied {@jls 5.1.8} followed by a
-     *     Java casting conversion {@jls 5.5} converting either directly to
+     *     a Java unboxing conversion is applied (JLS {@jls 5.1.8}) followed by a
+     *     Java casting conversion (JLS {@jls 5.5}) converting either directly to
      *     {@code dstType}, or, if {@code dstType} is {@code boolean},
      *     to {@code int}, which is then converted to either {@code true}
      *     or {@code false} depending on whether the least-significant-bit
-     *     is 1 or 0 respectively. If the runtime type of {@code value} is
-     *     not a primitive wrapper type a {@link ClassCastException} is thrown.
+     *     is 1 or 0 respectively.  If {@code value} is null, the zero value for
+     *     the {@code dstType} is returned.  Otherwise, a {@link ClassCastException}
```

Comment:
Is 'zero value' a well-defined term?

Reply:
Also, 'Otherwise' seems to refer to `If {@code value} is null` which I don't think is the intent? Maybe you could add another list item (`<li>`) With:

```
If {@code dstType} is a primitive type and the runtime type of {@code value} is <em>not</em> a primitive wrapper type, a {@link ClassCastException} is thrown
```

Reply:
It would be nice if the term was more well define, especially for `char` and `boolean`, but that doesn't have to be done in this patch.

I do think it would be good to split up the cases into separate list items though. (I made it way too convoluted the first time around). Like I said in my second comment, I don't think the current text works.